app/helpers/sessions_helper.rb

```diff
@@ -3,15 +3,36 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id])
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
```

Comment:
thử xem bỏ () có chạy ko

Reply:
và trả lời cho anh trình tự thực hiện của dòng này là gì, (thứ tự hoạt động của toán tử và method)

Reply:
Khi đang đăng nhập (cookie tạm thời - tạo bằng method session được lưu trong browser, session lưu trên máy chủ) thì sẽ nhảy vào if (cookie tạm thời và session mất khi tắt trình duyệt). current_user.id lấy giá trị id từ session.
Nếu không đăng nhập nhưng được lưu tài khoản (cookie dài hạn - tạo bằng method cookie lưu trong browser) thì nhảy vào elsif. current_user.id lấy giá trị id từ cookie dài hạn